### ISSUE TITLE
Introduce output type and renderer for background tasks

### ISSUE THREAD

**Issue by sourishkrout** [stateful/vscode-runme#287]
The idea is to have an "output rotation" of the last x lines inside the notebook while the background task runs in a dedicated terminal (which is already the case).

Using the GRPC Runner, we will finally have both in-notebook stdout/stderr access and a dedicated VS Code terminal.

The suggestion was to experiment using xterm.js to render the terminal output.

**Comment by mxsdev** [comment]
Discussed and this is what's needed before release:

 - [ ] Persistent terminal state (content, maybe cursor position)
 - [ ] Slideout functionality (want after #289 because this may have the same issue)
 - [ ] Put behind feature flag if we decide to merge/release pre-emptively
 - [ ] Focusing on terminal should not require two clicks (I think listen for iframe onFocus and direct that to terminal focus)
 
Nice-to-haves:

 - [ ] Theme colors totally consistent with VSCode's (need to deal with foreground over light background "mixed colors")
 - [ ] Avoid using `!important` in styling if possible
 - [ ] Font to be consistent with VSCode's terminal (both size and spacing)
 - [ ] Additional customizability options to user, styling, font, colors, etc
 - [ ] Scrollbar consistent with VSCode terminal
 - [ ] Offset padding on left and right of terminal (?? we can discuss this, matter of taste)

**Comment by degrammer** [comment]
Initial work for the new integrated runme terminal here https://github.com/stateful/vscode-runme/pull/308
